src/celeritas/optical/detail/PreGenAction.cu

```diff
@@ -0,0 +1,74 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/optical/detail/PreGenAction.cu
+//---------------------------------------------------------------------------//
+#include "PreGenAction.hh"
+
+#include <thrust/device_ptr.h>
+#include <thrust/execution_policy.h>
+#include <thrust/remove.h>
+
+#include "corecel/Assert.hh"
+#include "corecel/sys/ScopedProfiling.hh"
+#include "corecel/sys/Thrust.device.hh"
+#include "celeritas/global/ActionLauncher.device.hh"
+#include "celeritas/global/CoreParams.hh"
+#include "celeritas/global/CoreState.hh"
+#include "celeritas/global/TrackExecutor.hh"
+#include "celeritas/optical/CerenkovParams.hh"
+#include "celeritas/optical/OpticalPropertyParams.hh"
+#include "celeritas/optical/ScintillationParams.hh"
+
+#include "OpticalGenStorage.hh"
+#include "PreGenExecutor.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Generate optical distribution data post-step.
```

Comment:
Multiple files have this same doxygen description. Not sure if intentional, but we could be more clear on the specific function of each struct/class function.

Reply:
Nah that was just laziness, I'll try to make these more descriptive.